projects/dex-ui/src/components/Liquidity/allowance.ts

```diff
@@ -0,0 +1,37 @@
+import { ERC20Token } from "@beanstalk/sdk-core";
+import { BigNumber } from "ethers";
+
+const hasMinimumAllowance = async (walletAddress: string, spender: string, token: ERC20Token, mininumAllowance: BigNumber) => {
+  const existingAllowance = await token.getContract().allowance(walletAddress, spender);
+  
+  if (!existingAllowance) {
+    throw new Error("Unable to retrieve allowance");
```

Comment:
this error message is vague. how would we ever get into a null/undefined state? is it more likely that the call would fail and throw?

Also, forgot our previous convo, why use the contract method instead of the token's?

Reply:
First, I will remove this check. We can't get into that. It was a leftover from an earlier iteration.

Second, in this case, I can use token.getAllowance. In the other method, I have to use the contract method to specify the `gasLimit`.